foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/queries/ReportQueryResult.java

```diff
@@ -151,6 +124,44 @@ protected void buildResult(ReportQuery query, AbstractRecord row, Vector toManyD
         setResults(results);
     }
 
+    private Object processConstructorItem(ReportQuery query, AbstractRecord row, Vector toManyData, ConstructorReportItem constructorItem) {
+        // For constructor items need to process each constructor argument.
+        Class[] constructorArgTypes = constructorItem.getConstructorArgTypes();
+        int numberOfArguments = constructorItem.getReportItems().size();
+        Object[] constructorArgs = new Object[numberOfArguments];
+
+        for (int argumentIndex = 0; argumentIndex < numberOfArguments; argumentIndex++) {
+            ReportItem argumentItem = (ReportItem)constructorItem.getReportItems().get(argumentIndex);
```

Comment:
explicit cast may not be required here

Reply:
Nice catch, I missed this one! I was trying to remove all these explicit casts as part of this change too